default.nix

```diff
@@ -1,35 +1,41 @@
-{ rocksDBVersion ? "6.29.3" }:
+{ versions ? import ./nix/versions.nix
+, release ? true
+, name ? "aleph-node"
+, crates ? { "aleph-node" = []; }
+, runTests ? false
+, singleStep ? false
+, rustflags ? "-C target-cpu=native"
+, useCustomRocksDb ? false
+, rocksDbOptions ? { version = "6.29.3";
+                     useSnappy = false;
+                     patchVerifyChecksum = true;
+                     patchPath = ./nix/rocksdb.patch;
+                     enableJemalloc = true;
+                   }
+}:
 let
-  # this overlay allows us to use a specified version of the rust toolchain
-  rustOverlay =
-    import (builtins.fetchGit {
-      url = "https://github.com/mozilla/nixpkgs-mozilla.git";
-      rev = "f233fdc4ff6ba2ffeb1e3e3cd6d63bb1297d6996";
-    });
+  nixpkgs = versions.nixpkgs;
+  rustToolchain = versions.rustToolchain;
+  naersk = versions.naersk;
 
-  # pinned version of nix packages
-  # main reason for not using here the newest available version at the time or writing is that this way we depend on glibc version 2.31 (Ubuntu 20.04 LTS)
-  nixpkgs = import (builtins.fetchTarball {
-    url = "https://github.com/NixOS/nixpkgs/archive/2c162d49cd5b979eb66ff1653aecaeaa01690fcc.tar.gz";
-    sha256 = "08k7jy14rlpbb885x8dyds5pxr2h64mggfgil23vgyw6f1cn9kz6";
-  }) { overlays = [ rustOverlay ]; };
-
-  rustToolchain = with nixpkgs; ((rustChannelOf { rustToolchain = ./rust-toolchain; }).rust.override {
-    targets = [ "x86_64-unknown-linux-gnu" "wasm32-unknown-unknown" ];
-  });
+  # declares a build environment where C and C++ compilers are delivered by the llvm/clang project
+  # in this version build process should rely only on clang, without access to gcc
+  llvm = versions.llvm;
+  env = versions.stdenv;
+  llvmVersionString = "${nixpkgs.lib.getVersion env.cc.cc}";
 
   # we use a newer version of rocksdb than the one provided by nixpkgs
-  # we disable all compression algorithms and force it to use SSE 4.2 cpu instruction set
+  # we disable all compression algorithms, force it to use SSE 4.2 cpu instruction set and disable its `verify_checksum` mechanism
```

Comment:
I know it's a bit late comment, but what about we separate bumping RocksDB version into a separate change? This is to decouple the risk of significant rework of our build system, and changing one of crucial dependencies in a significant way (ie we provide a patch to a third party lib, this is imho quite invasive change). I'd be more comfortable to review this PR when RocksDB bump would be a separate PR, what do you think?